.github/workflows/c3-e2e-project-cleanup.yml

```diff
@@ -0,0 +1,30 @@
+# This workflow cleans up any leftover projects created by e2e runs.
+
+name: C3 E2E Project Cleanup
+on:
+  workflow_run:
+    workflows:
+      ["C3 E2E Tests", "C3 E2E Tests (Dependabot)", "C3 E2E (Quarantine)"]
+    types:
+      - completed
+env:
+  node-version: 18.17.1
+jobs:
+  cleanup:
+    name: "Cleanup Test Projects"
+    if: ${{ github.repository_owner == 'cloudflare' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Install Dependencies
+        uses: ./.github/actions/install-dependencies
+
+      - name: Cleanup E2E test projects
+        run: pnpm run --filter create-cloudflare test:e2e:cleanup
```

Comment:
Is this in case an e2e test crashes?
If not, why doesn't the test clean up after itself?